fix ~ touch: fix and test for windows

rivy opened the pull request:
- fix windows compilation
- enable testing windows
- re-categorize as "generic"

Arcterus commented on src/touch/touch.rs:
```diff
@@ -18,7 +18,9 @@ extern crate uucore;
 use std::fs::{self, File};
-use std::io::{self, Error};
+#[cfg(not(windows))]
+use std::io::{self};
```
This can just be `use std::io;`

Arcterus commented on src/touch/touch.rs:
```diff
@@ -52,7 +55,7 @@ macro_rules! to_timeval {
 
 #[cfg(unix)]
-fn set_symlink_times(p: &str, atime: FileTime, mtime: FileTime) -> io::Result<()> {
+fn set_symlink_file_times(p: &str, atime: FileTime, mtime: FileTime) -> io::Result<()> {
```
I don't think this function as actually necessary even on Unix.  `set_symlink_file_times()` in `filetime` seems to say it does the same thing.

rivy replied:
🛠 * both fixed

rivy commented:
@Arcterus , I'll review and update the code this weekend. Thanks for getting back to it.

Arcterus commented on src/touch/touch.rs:
```diff
@@ -42,6 +44,7 @@ macro_rules! local_tm_to_filetime(
     })
 );
 
+#[cfg(unix)]
```
Because the other function is now gone, this macro actually isn't used anywhere now, so it can be deleted too.

rivy replied:
👍

Arcterus commented on src/touch/touch.rs:
```diff
@@ -18,7 +18,9 @@ extern crate uucore;
-use std::io::{self, Error};
+#[cfg(not(windows))]
+use std::io;
+use std::io::{Error};
```
Sorry, I missed this one.  Can you remove the braces around `Error`?

rivy replied:
👍

rivy commented:
I squashed the code change and dead-code removal into a single commit.
And, I think I found and fixed all the code nits.
Local testing passes for windows and *nix.
Let me know if this passes muster.

Arcterus commented:
Looks good, thanks!
